### PR TITLE
f_op_kankyo 4/5 matched

### PR DESCRIPTION
**Pull request by Pheenoh**

### CODE REVIEW

**Comment by Pheenoh** [comment]
hold off approving/merging this one. realizing just now this has some dumb mistakes in it

**Comment by Pheenoh** [comment]
k should be good now

File: src/f_op/f_op_kankyo.cpp [hunk 1]
```diff
@@ -44,58 +24,93 @@ extern "C" void fpcMtd_Execute__FP20process_method_classPv();
...
-#pragma optimization_level 0
-#pragma optimizewithasm off
-static asm void fopKy_Execute(void* param_0) {
-    nofralloc
-#include "asm/f_op/f_op_kankyo/fopKy_Execute__FPv.s"
+static int fopKy_Execute(void* param_0) {
+    int ret;
+
+    if (dScnPly_c::isPause() && (!dComIfGp_isPauseFlag() || fpcM_GetName(param_0) == 0x15)) {
+        ret = fpcMtd_Execute(((process_method_class*)((kankyo_class*)param_0)->field_0xd8),param_0);
```
Comment (TakaRikka) [inline src/f_op/f_op_kankyo.cpp:48]: i wonder if using static_cast's helps these look cleaner. the C style casts look kinda crazy here. also any of the fpcM/fopAcM GetName funcs are getting one of the ProcName's from the d_procname enum, so the 0x15 should probably use the appropriate procname

File: src/f_op/f_op_kankyo.cpp [hunk 2]
```diff
@@ -44,58 +24,93 @@ extern "C" void fpcMtd_Execute__FP20process_method_classPv();
...
+    process_method_class* proc_mtd = (process_method_class*)leaf_mtd;
+    int ret = fpcMtd_Delete(proc_mtd,env);
+    fopDwTg_DrawQTo(&env->field_0xc4);
+    return ret;
 }
-#pragma pop
 
 /* ############################################################################################## */
 /* 80450CE8-80450CF0 0001E8 0004+04 1/1 0/0 0/0 .sbss            fopKy_KANKYO_TYPE */
 static u8 fopKy_KANKYO_TYPE[4 + 4 /* padding */];
```
Comment (TakaRikka) [inline src/f_op/f_op_kankyo.cpp:81]: can this be changed to int or does it cause issues?

**Review by TakaRikka — APPROVED** [review]

**Pull request merged**